rx/core/operators/distinct.py

```diff
@@ -47,7 +62,14 @@ def distinct(source: Observable) -> Observable:
         """
 
         def subscribe(observer, scheduler=None):
-            hashset = HashSet(comparer)
+            hashset = HashSet(comparer, lock=flushes.lock if flushes else None)
+
+            if flushes:
+                disposable = flushes.subscribe_(lambda _: hashset.flush(), scheduler=scheduler)
+                source.subscribe_(on_error=lambda _: disposable.dispose(),
```

Comment:
when a *flushes* observable is provided, the source observable will be subscribed twice instead on once. Do we want to dispose the flushes observable on completion/error or on source disposal ? Usually it is done on disposal ( the latter case°. In this case disposal can be modified by adding the disposables of flushes.subscribe and source.subscribe to a a CompositeDisposable, and return this disposable.

Reply:
Thanks! Updated